### ISSUE TITLE
Make it possible to insert an error message to the exception that is wrapped in ActionException

### ISSUE THREAD

**Issue by naonaonao316** [dakusui/actionunit#106]
**Background**
Although there are certain situations in which the root cause of the issue that makes exception happen in the project that inherits actionunit, it is not possible to insert a message to the exception when it is thrown, because there is no mechanism to have a message in the exception that is thrown within actionunit.
For instance, timeout exception is thrown at [InternalUtils](https://github.com/dakusui/actionunit/blob/e5787e050f9b2c043023ce62dabec36c3201a0e5/src/main/java/com/github/dakusui/actionunit/utils/InternalUtils.java#L40), which is wrapped in [ActionException](https://github.com/dakusui/actionunit/blob/5.2.x/src/main/java/com/github/dakusui/actionunit/exceptions/ActionException.java#L51).
It would be desirable for the message to be given to the exception to describe the why the it gets thrown in detail.

**TODO**
Make it possible to insert an error message to the exception that is wrapped in ActionException

**Comment by nobuoonodera** [comment]
**Request**
I want to know where the error happens.

**proposals**
1. I hope to contain the action name where the error happens. 
2. I hope to display a part of the action tree including the action having a problem.  Because the action name is '(nop)' sometimes. This helps the user find the issue.

**Comment by dakusui** [comment]
After a offline talk with @nobuoonodera , the thing here is:

- If something has already failed, "timeout" won't happen from the beginning (because the exception will make the timeout structure fail immediately)
- However, in an idiom where timeout structure which has retry structure in it (, which is so often used, ) the exception on which retry is performed will be masked. Because it will be retried and not yet failed.
- In this structure, if the timeout really happens because retries take too long time, what the user want to know is the masked exception. Not the time out exception itself.

**Comment by dakusui** [comment]
I've marked this "enhancement" because this is basically a designed behavior.

**Comment by dakusui** [comment]
You will see following message, when on-goring retry action times out.
```
com.github.dakusui.actionunit.exceptions.ActionTimeOutException: Action: <timeout in 1 [seconds]>; 
throw an Exception
----
RETRYING
	com.github.dakusui.actionunit.ut.actions.TimeoutTest.lambda$givenNotEndingRetryActionInside$whenTimeoutOutside$thenTimeoutsAppropriately$1(TimeoutTest.java:90)
	com.github.dakusui.actionunit.actions.Leaf$1.lambda$runnable$0(Leaf.java:28)
	com.github.dakusui.actionunit.visitors.ActionPerformer.visit(ActionPerformer.java:27)
	com.github.dakusui.actionunit.actions.Leaf.accept(Leaf.java:20)
	com.github.dakusui.actionunit.visitors.SimpleActionPerformer.callAccept(SimpleActionPerformer.java:18)
	com.github.dakusui.actionunit.visitors.ActionPerformer.visit(ActionPerformer.java:31)
```

**Comment by dakusui** [comment]
I've now released this fix as 5.2.12.
Closing this ticket.
